sentry-android-core/src/main/java/io/sentry/android/core/AppStartState.java

```diff
@@ -0,0 +1,68 @@
+package io.sentry.android.core;
+
+import android.os.SystemClock;
+import io.sentry.DateUtils;
+import java.util.Date;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class AppStartState {
+
+  private static final @NotNull AppStartState instance = new AppStartState();
+
+  private @Nullable Long appStart;
+  private @Nullable Long appStartEnd;
+  private @Nullable Boolean coldStart;
+  private boolean sentStartMetric = false;
+  private @Nullable Date appStartTime;
+  private volatile boolean appStartSet = false;
+
+  private AppStartState() {}
+
+  static @NotNull AppStartState getInstance() {
+    return instance;
+  }
+
+  void setAppStartEnd(final @Nullable Long appStartEnd) {
+    this.appStartEnd = appStartEnd;
+  }
+
+  @Nullable
+  Long getAppStartInterval() {
+    if (appStart == null || appStartEnd == null || coldStart == null) {
+      return null;
+    }
+    return appStartEnd - appStart;
+  }
+
+  boolean getColdStart() {
+    return Boolean.TRUE.equals(coldStart);
+  }
+
+  void setColdStart(final boolean coldStart) {
+    this.coldStart = coldStart;
+  }
+
+  boolean isSentStartMetric() {
+    return sentStartMetric;
+  }
+
+  void setSentStartUp() {
+    this.sentStartMetric = true;
+  }
+
+  @Nullable
+  Date getAppStartTime() {
+    return appStartTime;
+  }
+
+  synchronized void setAppStartTime() {
+    if (appStartSet) {
+      return;
+    }
+    final long millis = SystemClock.uptimeMillis();
```

Comment:
`m`: Why do we use `SystemClock.uptimeMillis()` and not the current date? Can we add the reason somewhere in a comment and also document this in the setter of appStartEnd.

Reply:
to measure intervals, you have to use `SystemClock.uptimeMillis()`, the current time might be changed by the user or by the phone provider, timezone etc